src/libcprover-cpp/verification_result.cpp

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) 2023 Fotis Koutoulakis for Diffblue Ltd.
+
+/// \file
+/// Interface for the various verification engines providing results
+//  in a structured form.
+
+#include "verification_result.h"
+
+#include <util/invariant.h>
+#include <util/make_unique.h>
+
+#include <goto-checker/properties.h>
+
+#include <algorithm>
+#include <string>
+#include <vector>
+
+class verification_resultt::verification_result_implt
```

Comment:
:thinking: I think you can just make this class a struct, delete all the member functions and just use the fields directly. It would be far more straight-forward.

Reply:
I disagree with your reasoning. "impl" means implementation detail; it is in the name. It shouldn't be intended to be part of the exposed API. Therefore the using a class to add accessibility restrictions to something which is already private by virtual of being declared in the `.cpp` only is overkill.

This is not a blocking comment, just an observation. So it can be left as-is for the moment if you prefer.